src/Ball.js

```diff
@@ -74,15 +100,27 @@ export class Ball {
         otherBall.vy = ay
 
         // both has same state, so nothing to do
-        if (this.state === state) return
+        if (this.state === otherBallState) return
         // if any is recovered, then nothing happens
-        if (this.state === STATES.recovered || state === STATES.recovered) return
+        if (this.state === STATES.recovered || otherBallState === STATES.recovered) return
         // then, if some is infected, then we make both infected
-        if (this.state === STATES.infected || state === STATES.infected) {
-          if (this.hasAppInstalled && otherBall.hasAppInstalled) return
-          this.state = otherBall.state = STATES.infected
-          RUN.results[STATES.infected]++
-          RUN.results[STATES.well]--
+        if (this.state === STATES.infected || otherBallState === STATES.infected) {
+          if (this.state === STATES.well) {
+            this.state = STATES.incubating
+            RUN.results[STATES.incubating]++
+            RUN.results[STATES.well]--
+            if (this.hasAppInstalled && otherBall.hasAppInstalled) {
+              this.hasMovement = false
```

Comment:
please, add a comment to explain why the person stops moving.

Reply:
Resolved with a private method with a meaningful name and a comment